.github/workflows/build-test-publish-images.yml

```
@@ -31,6 +31,28 @@ permissions:
   statuses: none
 
 jobs:
+  pr-builder:
+    if: ${{ !cancelled() && inputs.build_type == 'pull-request' }}
+    needs:
+      - checks
+      - compute-matrix
+      - build
+      - test
+      - delete-temp-images
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/pr-builder.yaml@branch-24.10
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
```

Might as well delay updating by one more minor version.
```suggestion
          python-version: "3.11"
```

---

Yeah was wondering if we should do this

Is there value in running this with the oldest Python we support?

---

This version only affects `pre-commit` checking of the glue code in this repo, like https://github.com/rapidsai/docker/blob/branch-24.10/context/test_notebooks.py ... not any Python code that ever imports anything from RAPIDS.

And it's only in the diff here because I'm moving things around.

But while we're now talking about it... I think it would be safe (and one less source of maintenance burden) to remove this `setup-python` entirely and just take whatever version ships on the `ubuntu-latest` VM image from GitHub Actions.

---

> And it's only in the diff here because I'm moving things around.

Yeah was wondering for this reason whether this should be its own issue for separate follow up. That said, no strong feelings either way

---

I pushed a commit just removing `setup-python` and re-using whatever Python and `pip` is around in the `ubuntu-latest` image from GitHub Actions.

Seems like it worked well: https://github.com/rapidsai/docker/actions/runs/10408813338/job/28827015511?pr=708

I think we should do that until we encounter a version-related issue. The set of `pre-commit` checks here is pretty minimal and only running on a bit of glue code... I think the risk of issues is pretty low, and worth it in exchange for having one less place to update as our Python support matrix shifts.